### PR TITLE
ROX-9640: Port Compliance over to Postgres

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Work in progress PR for porting compliance over to Postgres. This will initially be a naive port

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Just trying to build it

### CODE REVIEW

File: central/compliance/datastore/internal/store/postgres/store.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+	}
+	return types.ResultsWithStatus{
+		LastSuccessfulResults: results,
+	}, nil
+}
+
+func (s *storeImpl) GetLatestRunResults(ctx context.Context, clusterID, standardID string, flags types.GetFlags) (types.ResultsWithStatus, error) {
+	metadataBatch, err := s.getLatestRunMetadata(ctx, clusterID, standardID)
+	if err != nil {
+		return types.ResultsWithStatus{}, nil
```
Comment (dashrews78) [inline central/compliance/datastore/internal/store/postgres/store.go:138]: Shouldn't we return err here?  Later when this thing is called it checks if it returns err.

File: central/compliance/datastore/internal/store/postgres/store.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+	}
+	if metadataBatch.LastSuccessfulRunMetadata == nil {
+		return types.ResultsWithStatus{
+			FailedRuns: metadataBatch.FailedRunsMetadata,
+		}, nil
+	}
+	results, err := s.getResultsFromMetadata(ctx, metadataBatch.LastSuccessfulRunMetadata, flags)
+	if err != nil {
+		return types.ResultsWithStatus{}, nil
```
Comment (dashrews78) [inline central/compliance/datastore/internal/store/postgres/store.go:147]: Shouldn't we return err here?  Later when this thing is called it checks if it returns err.

File: central/compliance/datastore/internal/store/postgres/store.go (hunk 1 above)
Comment (connorgorman) [reply central/compliance/datastore/internal/store/postgres/store.go:138]: yikes, thanks

File: central/compliance/datastore/internal/store/postgres/store.go (hunk 2 above)
Comment (connorgorman) [reply central/compliance/datastore/internal/store/postgres/store.go:147]: yikes, thanks

File: central/compliance/datastore/internal/store/postgres/store.go [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+func (s *storeImpl) GetLatestRunResultsBatch(ctx context.Context, clusterIDs, standardIDs []string, flags types.GetFlags) (map[compliance.ClusterStandardPair]types.ResultsWithStatus, error) {
+	pairsToResults := make(map[compliance.ClusterStandardPair]types.ResultsWithStatus)
+	for _, clusterID := range clusterIDs {
+		for _, standardID := range standardIDs {
+			results, err := s.GetLatestRunResults(ctx, clusterID, standardID, flags)
+			if err != nil {
+				return nil, err
+			}
+			pairsToResults[compliance.ClusterStandardPair{ClusterID: clusterID, StandardID: standardID}] = results
```
Comment (dashrews78) [inline central/compliance/datastore/internal/store/postgres/store.go:163]: This is either a nit OR an opportunity for me to learn something.  Why do you set the map like this here but down below use the NewPair down below?
`results[compliance.NewPair(clusterID, standardID)] = metadata`

Comment (connorgorman) [reply central/compliance/datastore/internal/store/postgres/store.go:163]: because I messed up :)

**Review by dashrews78 — APPROVED** [review]
LGTM, only comment outstanding as I type is that tiny nit.

**Comment by connorgorman** [comment]
RocksDB:
Screen Shot 2022-06-14 at 2 50 45 PM

Postgres:
Screen Shot 2022-06-14 at 3 35 20 PM

Aggregation storage optimizations were turned off due to SAC implications so I have not implemented them unnecessarily. If we want to turn them back on, we will have to reconsider. In any case, we would probably want to optimize aggregations and calculations in Postgres anyways

**Pull request merged**